mteb/tasks/Clustering/multilingual/WikiClusteringP2P.py

```diff
@@ -48,3 +55,82 @@ class WikiClusteringP2P(AbsTaskClustering, MultilingualTask):
         n_samples={"test": 71680},
         avg_character_length={"test": 625.3},
     )
+
+
+class WikiClusteringFastP2P(AbsTaskClusteringFast, MultilingualTask):
+    metadata = TaskMetadata(
+        name="WikiClusteringFastP2P",
+        description="Clustering of wikipedia articles inspired by BlubrbsClusteringP2P. Labels are taken from top-level categories of the respective languages (e.g., https://lv.wikipedia.org/wiki/Kategorija:Pamatkategorijas).",
+        reference="https://github.com/Rysias/wiki-clustering",
+        dataset={
+            "path": "ryzzlestrizzle/multi-wiki-clustering-p2p",
+            "revision": "d4d92f8f28be71035be6a96bdfd4e200cf62faa8",
+        },
+        type="Clustering",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=_LANGUAGES,
+        main_score="v_measure",
+        date=("2001-01-15", "2024-04-15"),
+        form=["written"],
+        domains=["Encyclopaedic"],
+        task_subtypes=["Thematic clustering"],
+        license="cc-by-sa-3.0",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="",  # None exists
+        n_samples={"test": 2048},
+        avg_character_length={"test": 625.3},
+    )
+
+    def dataset_transform(self):
+        ds = dict()
+        for lang in self.hf_subsets:
+            labels = []
+            sentences = []
+            ds[lang] = dict()
+            lang_dict = dict()
+            for split in self.metadata.eval_splits:
+                labels.extend(
+                    list(
```

Comment:
You shouldn't need `list()` here. Extend takes an iterable anyway. You're unnecessarily using memory and evaluating prematurely this way.

Reply:
Good call! removing `list()`